R/pkg/tests/fulltests/test_sparkSQL.R

```diff
@@ -2558,7 +2558,7 @@ test_that("join(), crossJoin() and merge() on a DataFrame", {
   error_msg <- paste("joinType must be one of the following types:",
                  "'inner', 'cross', 'outer', 'full', 'fullouter', 'full_outer',",
                  "'left', 'leftouter', 'left_outer', 'right', 'rightouter', 'right_outer',",
-                 "'semi', 'leftsemi', 'left_semi', 'anti', 'leftanti' or 'left_anti'.")
+                 "'semi', 'left_semi', 'leftsemi', 'anti', 'left_anti', 'leftanti'")
```

Comment:
You can also 'fix' the order of the list in the original code to be consistent with the test instead. It's really minor but the order in the test looks more consistent: leftfoo, left_foo

Reply:
OK. I picked between the order in the error message and that in the `%in%` test a few lines above arbitrarily. I'll switch it around. The test will still change slightly since the error is easier to write without being too careful about the `or` at the end.